compiler/src/dotty/tools/backend/sjs/JSConstructorGen.scala

```diff
@@ -0,0 +1,376 @@
+package dotty.tools.backend.sjs
+
+import org.scalajs.ir
+import org.scalajs.ir.{Position, Trees => js, Types => jstpe}
+import org.scalajs.ir.Names._
+import org.scalajs.ir.OriginalName.NoOriginalName
+
+import JSCodeGen.UndefinedParam
+
+object JSConstructorGen {
+
+  /** Builds one JS constructor out of several "init" methods and their
+   *  dispatcher.
+   *
+   *  This method and the rest of this file are copied verbatim from `GenJSCode`
+   *  for scalac, since there is no dependency on the compiler trees/symbols/etc.
+   *  We are only manipulating IR trees and types.
+   *
+   *  The only difference is the two parameters `overloadIdent` and `reportError`,
+   *  which are added so that this entire file can be even more isolated.
```

Comment:
Given the verbatim copy from Scala 2, it is probably completely useless to review this file.